substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVTypeRecord.java

```diff
@@ -235,4 +411,795 @@ public boolean equals(Object obj) {
+            int pos = CVUtil.putShort(count, buffer, initialPos);
+            pos = CVUtil.putShort(propertyAttributes, buffer, pos);
+            pos = CVUtil.putInt(fieldIndex, buffer, pos);
+            pos = CVUtil.putInt(derivedFromIndex, buffer, pos);
+            pos = CVUtil.putInt(vshapeIndex, buffer, pos);
+            pos = CVUtil.putLfNumeric(size, buffer, pos);
+            String fixedName = CVNames.typeNameToCodeViewName(className);
+            pos = CVUtil.putUTF8StringBytes(fixedName, buffer, pos);
+            if (hasUniqueName()) {
+                assert uniqueName != null;
+                pos = CVUtil.putUTF8StringBytes(uniqueName, buffer, pos);
+            }
+            return pos;
+        }
+
+        boolean isForwardRef() {
+            return (propertyAttributes & ATTR_FORWARD_REF) != 0;
+        }
+
+        @SuppressWarnings("unused")
+        public boolean hasUniqueName() {
+            return (propertyAttributes & ATTR_HAS_UNIQUENAME) != 0;
+        }
+
+        protected String toString(String lfTypeStr) {
+            return String.format("%s 0x%04x count=%d attr=0x%x(%s) fld=0x%x super=0x%x vshape=0x%x size=%d %s%s", lfTypeStr, getSequenceNumber(), count, propertyAttributes,
+                            propertyString(propertyAttributes), fieldIndex, derivedFromIndex,
+                            vshapeIndex, size, className, uniqueName != null ? " (" + uniqueName + ")" : "");
+        }
+
+        @Override
+        public String toString() {
+            return toString("LF_CLASS");
+        }
+
+        @Override
+        public int hashCode() {
+            int h = type;
+            h = 31 * h + count;
+            h = 31 * h + propertyAttributes;
+            h = 31 * h + fieldIndex;
+            h = 31 * h + derivedFromIndex;
+            h = 31 * h + (int) size;
+            h = 31 * h + className.hashCode();
+            if (uniqueName != null) {
+                h = 31 * h + uniqueName.hashCode();
+            }
+            h = 31 * h + vshapeIndex;
+            return h;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (!super.equals(obj)) {
+                return false;
+            }
+            CVClassRecord other = (CVClassRecord) obj;
+            return this.count == other.count && this.propertyAttributes == other.propertyAttributes && this.fieldIndex == other.fieldIndex && this.derivedFromIndex == other.derivedFromIndex &&
+                            this.size == other.size && this.className.equals(other.className) && (this.uniqueName != null ? this.uniqueName.equals(other.uniqueName) : other.uniqueName == null) &&
+                            this.vshapeIndex == other.vshapeIndex;
+        }
+    }
+
+    static final class CVFieldListRecord extends CVTypeRecord {
+
+        static final int INITIAL_CAPACITY = 10;
+
+        /* Size includes type field and record length field. */
+        private int estimatedSize = Short.BYTES * 2;
```

Comment:
I'd rather keep this as it was. While the value is technically the same in both cases, it's not because the size includes a length field, but because of the alignment. 
```suggestion
        /* Size includes type field but not record length field. */
        private int estimatedSize = CVUtil.align4(Short.BYTES);
```

Reply:
Reverted.